pkg/agent/baker.go

```diff
@@ -607,3 +625,35 @@ func getContainerServiceFuncMap(cs *api.ContainerService) template.FuncMap {
 		},
 	}
 }
+
+func getDockerConfig(cs *api.ContainerService, hasGPU bool) (string, error) {
+	var overrides []func(*DockerConfig) error
+
+	if hasGPU {
+		overrides = append(overrides, DockerNvidiaOverride)
+	}
+
+	val, err := GetDockerConfig(cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig, overrides)
```

Comment:
We need to vendor the aks-engine code with your change

Reply:
Acknowledged, saw you already looked at the existing PR I opened to address that. Thanks!